src/researchhub_document/filters.py

```diff
@@ -50,227 +72,163 @@ class UnifiedDocumentFilter(filters.FilterSet):
     )
-    bounties = filters.ChoiceFilter(
-        field_name="bounties",
-        method="bounty_type_filter",
-        choices=BOUNTY_CHOICES,
-    )
 
     class Meta:
         model = ResearchhubUnifiedDocument
         fields = ["hub_id", "ordering", "subscribed_hubs", "type"]
 
+    def _map_tag_to_document_filter(self, value):
+        if value == "closed":
+            return "bounty_closed", True
+        elif value == "expired":
+            return "bounty_expired", True
+        elif value == "open":
+            return "bounty_open", True
+        elif value == "unanswered":
+            return "answered", False
+        return value, True
+
     def document_type_filter(self, qs, name, value):
         value = value.upper()
+        selects = (
+            "paper",
+            "paper__uploaded_by",
+            "paper__uploaded_by__author_profile",
+            "hypothesis",
+            "hypothesis__created_by",
+            "hypothesis__created_by__author_profile",
+        )
+        prefetches = (
+            "hubs",
+            "paper__hubs",
+            "paper__purchases",
+            "paper__figures",
+            "posts",
+            "posts__created_by",
+            "posts__created_by__author_profile",
+            "posts__purchases",
+            "reviews",
+            "related_bounties",
+        )
 
         if value == PAPER:
             qs = qs.filter(document_type=PAPER)
+            selects = (
+                "paper",
+                "paper__uploaded_by",
+                "paper__uploaded_by__author_profile",
+            )
+            prefetches = (
+                "hubs",
+                "paper",
+                "reviews",
+                "related_bounties",
+                "paper__hubs",
+                "paper__figures",
+                "paper__purchases",
+            )
         elif value == POSTS:
             qs = qs.filter(document_type__in=[DISCUSSION, ELN])
+            selects = []
+            prefetches = [
+                "hubs",
+                "reviews",
+                "related_bounties",
+                "posts",
+                "posts__created_by",
+                "posts__created_by__author_profile",
+                "posts__purchases",
+            ]
         elif value == QUESTION:
             qs = qs.filter(document_type=QUESTION)
+            selects = []
         elif value == HYPOTHESIS:
-            qs = qs.filter(document_type=HYPOTHESIS).prefetch_related(
-                "hypothesis__votes", "hypothesis__citations"
+            qs = qs.filter(document_type=HYPOTHESIS)
+            selects = (
+                "hypothesis",
+                "hypothesis__created_by",
+                "hypothesis__created_by__author_profile",
+            )
+            prefetches = (
+                "reviews",
+                "related_bounties",
+                "hypothesis__votes",
+                "hypothesis__citations",
+            )
+        elif value == BOUNTY:
```

Comment:
Bundling bounty with uniDoc filter makes me feel a little uneasy because bounties aren't really unified documents. Maybe I'll revisit this when i tackle some of these during techdebt. 

Just making a comment here for reference